malt/data/dataset.py

```diff
@@ -69,7 +74,7 @@ def __getitem__(self, idx):
         if isinstance(idx, int):
             return self.points[idx]
         elif isinstance(idx, str):
-            return self.lookup[idx]
+            return self.__class__(points=self.lookup[idx])
```

Comment:
@yuanqing-wang the fundamental issue is that a dataset can contain multiple datapoints with the same smiles. what then is the appropriate behavior? At least, it should return a list of datapoints. But it seems to me that it may be appropriate to wrap that in a dataset, the same way that if we index several points by `int`s, we also return a new Dataset object.